CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -87,8 +87,39 @@ public class Child {
   /// This is used to e.g. describe the child if all of its tokens are missing in the source file.
   public let nameForDiagnostics: String?
 
-  /// A doc comment describing the child.
-  public let documentation: SwiftSyntax.Trivia
+  /// A summary of a doc comment describing the child. Full docc comment for the
+  /// child is available in ``Child/documentation``, and includes detailed list
+  /// of possible choices for the child if it's a token kind.
+  public let documentationSummary: SwiftSyntax.Trivia
+
+  /// A docc comment describing the child, including the trivia provided when
+  /// initializing the ``Child``, and the list of possible token choices inferred automatically.
+  public var documentation: SwiftSyntax.Trivia {
+    if case .token(let choices, _, _) = kind {
+      let grammar = GrammarGenerator()
+
+      let choices =
+        choices.count == 1
+        ? "kind: \(grammar.grammar(for: choices.first!))"
+        : """
+        kinds:
+        \(choices.map { " - \(grammar.grammar(for: $0))" }.joined(separator: "\n"))
+        """
+
+      let tokenChoicesTrivia = SwiftSyntax.Trivia.docCommentTrivia(
+        from: """
+          ### Tokens
+
+          For syntax trees generated by the parser, this is guaranteed to be one of the following \(choices)
```

Comment:
Oh, sorry, maybe I should have been cleaner. In the singular, it should be “this is guaranteed to be the following kind” (“one of” should only be there in the plural form). Overall I think it might be easiest to just have two completely separate branches for the singular and plural case.